backend/checkmach.ml

```diff
@@ -110,10 +125,19 @@ module Witnesses : sig
 end = struct
   include Set.Make (Witness)
 
-  (* CR gyorsh: consider using Flambda_backend_flags.checkmach_details_cutoff to
-     limit the size of this set. The downside is that it won't get tested as
-     much. Only keep witnesses for functions that need checking. *)
-  let join = union
+  let print ppf t = Format.pp_print_seq Witness.print ppf (to_seq t)
+
+  let cutoff t ~n = take_first_n t n ~fold ~remove ~cardinal
+
+  let join t1 t2 =
+    let res = union t1 t2 in
```

Comment:
That could be changed in a follow-up, but
it feels a bit weird to first compute the
union unconstrained to then remove a
number of values if the union happens to
be too large. Would repeated additions be
bad in terms of complexity?

Reply:
(If it is important to select the first `n` values
according to the order, we could iterate
over a couple of sequences until we reach
the threshold.)

Reply:
Added a CR.